### ISSUE TITLE
Weird transparency/plot size interaction in svgs

### ISSUE THREAD

**Issue by benureau** [bokeh/bokeh#7883]
bokeh 0.12.15
python 3.6.4
macOS 10.13.4
npm@5.6.0
phantomjs@2.1.7

#### Description of expected behavior and the observed behavior

When exporting to svgs, transparency of the background is respected if the plot width and height is 500, but not in (some) other cases, for instance 600 or 400.  

#### Complete, minimal, self-contained example code that reproduces the issue

```
from bokeh import plotting
from bokeh.io import export_png, export_svgs

fig = plotting.figure()
fig.background_fill_color = None
fig.border_fill_color     = None
fig.line(range(3), range(3))
export_png(fig, filename='fig.png')
fig.output_backend = 'svg'
export_svgs(fig, filename='fig.svg')


fig2 = plotting.figure(plot_width=500, plot_height=500)
fig2.background_fill_color = None
fig2.border_fill_color     = None
fig2.line(range(3), range(3))
export_png(fig2, filename='fig2.png')
fig2.output_backend = 'svg'
export_svgs(fig2, filename='fig2.svg')


fig3 = plotting.figure(plot_width=400, plot_height=400)
fig3.background_fill_color = None
fig3.border_fill_color     = None
fig3.line(range(3), range(3))
export_png(fig3, filename='fig3.png')
fig3.output_backend = 'svg'
export_svgs(fig3, filename='fig3.svg')
```

So in these examples, fig.png, fig2.png and fig3.png should all have a transparent background. However, while fig2.svg has a transparent background, fig.svg and fig3.svg don't. One can check that by looking for the string `<rect fill="#FFFFFF"` in the svg file, present in fig.svg and fig3.svg but not in fig2.svg.

**Comment by bryevdv** [comment]
@canavandl any quick ideas?

**Comment by bryevdv** [comment]
possibly related to #6775

**Comment by bryevdv** [comment]
I also tried setting alphas to zero but I don't think that had an effect. This version sets colors with the zero alpha, it's easy to see there is an extra `rect fill` (uppercase) that did not come from the plot settings 
```
from bokeh import plotting
from bokeh.io import export_png, export_svgs

fig = plotting.figure()
fig.background_fill_color = "#ff0000"
fig.border_fill_color     = "#00ffff"
fig.background_fill_alpha = 0
fig.border_fill_alpha     = 0
fig.line(range(3), range(3))
export_png(fig, filename='fig.png')
fig.output_backend = 'svg'
export_svgs(fig, filename='fig.svg')


fig2 = plotting.figure(plot_width=500, plot_height=500)
fig2.background_fill_color = "#ff0000"
fig2.border_fill_color     = "#00ffff"
fig2.background_fill_alpha = 0
fig2.border_fill_alpha     = 0
fig2.line(range(3), range(3))
export_png(fig2, filename='fig2.png')
fig2.output_backend = 'svg'
export_svgs(fig2, filename='fig2.svg')


fig3 = plotting.figure(plot_width=400, plot_height=400)
fig3.background_fill_color = "#ff0000"
fig3.border_fill_color     = "#00ffff"
fig3.background_fill_alpha = 0
fig3.border_fill_alpha     = 0
fig3.line(range(3), range(3))
export_png(fig3, filename='fig3.png')
fig3.output_backend = 'svg'
export_svgs(fig3, filename='fig3.svg')
```

**Comment by bryevdv** [comment]
Let's leave this for @brendancol et al

**Comment by mattpap** [comment]
This seems to have been fixed at some point (possibly in PR #10402), e.g. when combined using:
```html
<html>
  <head>
  </head>
  <body style="background-color: purple;">
    <img src="./fig.svg"></img>
    <img src="./fig2.svg"></img>
    <img src="./fig3.svg"></img>
  </body>
</html>
```
it gives
![image](https://user-images.githubusercontent.com/27475/127865016-4fcbb56c-8142-4c64-ac2b-1c2007ba8215.png)

**Comment by bryevdv** [comment]
Running the old code I agree this seems to be fixed.
